File: bindings/java/src/main/java/com/ainos/sdk/models/AinosConnectionException.java
package com.ainos.sdk.models;

/**
 * Exception thrown when the SDK cannot establish or maintain a connection
 * to the Ainos daemon.
 * <p>
 * This can occur during initial connection, after a socket timeout,
 * or when the remote end closes the connection unexpectedly.
 */
public class AinosConnectionException extends AinosException {

    private static final long serialVersionUID = 1L;

    public AinosConnectionException(String message) {
        super(message);
    }

    public AinosConnectionException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: bindings/csharp/AinosSdk/Models/AinosConnectionException.cs
namespace AinosSdk.Models;

/// <summary>
/// Raised when the SDK cannot establish or maintain a connection to the daemon.
/// </summary>
public class AinosConnectionException : AinosException
{
    /// <summary>
    /// The host that was being connected to.
    /// </summary>
    public string? Host { get; }

    /// <summary>
    /// The port that was being connected to.
    /// </summary>
    public int? Port { get; }

    /// <summary>
    /// Whether the connection was lost after being established.
    /// </summary>
    public bool WasConnected { get; }

    public AinosConnectionException()
    {
    }

    public AinosConnectionException(string message) : base(message)
    {
    }

    public AinosConnectionException(string message, Exception innerException) : base(message, innerException)
    {
    }

    public AinosConnectionException(string message, string host, int port)
        : base($"Cannot connect to {host}:{port} — {message}")
    {
        Host = host;
        Port = port;
    }

    public AinosConnectionException(string message, string host, int port, bool wasConnected)
        : base(message)
    {
        Host = host;
        Port = port;
        WasConnected = wasConnected;
    }
}